File: 1894-merge-strings-alternately/1894-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        max_len = max(len(word1), len(word2))
        # strings are immutable in python. 
        # so it is faster to use a list and join it to make a string
        res = []

        for i in range(max_len):
            if i < len(word1):
                res.append(word1[i])
            if i < len(word2):
                res.append(word2[i])
        
        return ''.join(res)

File: 1894-merge-strings-alternately/1894-merge-strings-alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {

    const res = [];

    const maxLen = Math.max(word1.length, word2.length);

    for (let i = 0; i < maxLen; i++) {
        if (i < word1.length) res.push(word1[i]);
        if (i < word2.length) res.push(word2[i]);
    };

    return res.join('');
    
};